include/wil/network.h

```diff
@@ -0,0 +1,1338 @@
+    // - hence the common use of "host-to-network" and "network-to-host" APIs, e.g. htons(), htonl(), ntohs(), ntohl()
+    //
+    // Socket APIs that accept a socket address will accept 2 fields:
+    // - the sockaddr* (the address of the derived sockaddr type, cast down to a sockaddr*)
+    // - the length of the 'derived' socket address structure referenced by the sockaddr*
+    //
+    // Commonly used sockaddr* types that are using with TCPIP networking:
+    //
+    // sockaddr_storage / SOCKADDR_STORAGE
+    //   - a sockaddr* derived type that is guaranteed to be large enough to hold any possible socket address (not just TCPIP related)
+    // sockaddr_in / SOCKADDR_IN
+    //   - a sockaddr* derived type designed to contain an IPv4 address and port number
+    // sockaddr_in6 / SOCKADDR_IN6
+    //   - a sockaddr* derived type designed to contain an IPv6 address, port, scope id, and flow info
+    // SOCKADDR_INET
+    //   - a union of sockaddr_in and sockaddr_in6 -- i.e., large enough to contain any TCPIP IPv4 or IPV6 address
+    // in_addr / IN_ADDR
+    //   - the raw address portion of a sockaddr_in
+    // in6_addr / IN6_ADDR
+    //   - the raw address portion of a sockaddr_in6
+    //
+    // SOCKET_ADDRESS
+    //   - not a derived sockaddr* type
+    //   - a structure containing both a sockaddr* and its length fields, returned from some networking functions
+    //
+
+    // declaring char-arrays large enough for any IPv4 or IPv6 address + optional fields
+    static_assert(INET6_ADDRSTRLEN > INET_ADDRSTRLEN);
+    typedef WCHAR socket_address_wstring[INET6_ADDRSTRLEN];
+    typedef CHAR socket_address_string[INET6_ADDRSTRLEN];
+
+    class socket_address final
+    {
+    public:
+        constexpr socket_address() WI_NOEXCEPT = default;
+        explicit socket_address(ADDRESS_FAMILY family) WI_NOEXCEPT;
+        template <typename T>
+        explicit socket_address(_In_reads_bytes_(addr_size) const SOCKADDR* addr, T addr_size) WI_NOEXCEPT;
+        explicit socket_address(const SOCKADDR_IN*) WI_NOEXCEPT;
+        explicit socket_address(const SOCKADDR_IN6*) WI_NOEXCEPT;
+        explicit socket_address(const SOCKADDR_INET*) WI_NOEXCEPT;
+        explicit socket_address(const SOCKET_ADDRESS*) WI_NOEXCEPT;
+        explicit socket_address(const IN_ADDR*, unsigned short port = 0) WI_NOEXCEPT;
+        explicit socket_address(const IN6_ADDR*, unsigned short port = 0) WI_NOEXCEPT;
+#if defined(WIL_ENABLE_EXCEPTIONS)
+        explicit socket_address(PCWSTR, unsigned short port = 0);
+#endif
+
+        ~socket_address() = default;
+        socket_address(const socket_address&) WI_NOEXCEPT = default;
+        socket_address& operator=(const socket_address&) WI_NOEXCEPT = default;
+        socket_address(socket_address&&) WI_NOEXCEPT = default;
+        socket_address& operator=(socket_address&&) WI_NOEXCEPT = default;
+
+        bool operator==(const socket_address&) const WI_NOEXCEPT;

```

Comment:
```suggestion
#endif

        bool operator==(const socket_address&) const WI_NOEXCEPT;
```
That's already the default